tools/hicup/hicup_mapper.xml

```diff
@@ -4,55 +4,81 @@
         <import>hicup_macros.xml</import>
     </macros>
     <expand macro="requirements_hicup" />
-    <expand macro="stdio" />
-
-    <command><![CDATA[
+    <command detect_errors="exit_code"><![CDATA[
         BOWTIE_PATH_BASH="\$(which bowtie2)" &&
         #set index_path = ''
         #if str($reference_genome.source) == "history":
-           bowtie2-build "$reference_genome.own_file" genome &&
-           ln -s "$reference_genome.own_file" genome.fa &&
+           bowtie2-build '$reference_genome.own_file' genome &&
+           ln -s '$reference_genome.own_file' genome.fa &&
            #set index_path = 'genome'
         #else:
            #set index_path = $reference_genome.index.fields.path
         #end if
-        hicup_mapper
-        --index '$index_path' 
-        --bowtie2 \$BOWTIE_PATH_BASH
 
-        $input_first_sequence $input_second_sequence
+        ##Dealing with fastq and fastq.gz
+        input1='$input_first_sequence' &&
+        input2='$input_second_sequence' &&
+        trunc_result_1=\$(echo '$input_first_sequence' | rev | cut -d'/' -f1 | rev) &&
```

Comment:
This is just to create a filename, right? Does it need to be a filename? Can we use the element_identifier (esacped) https://github.com/galaxyproject/tools-iuc/blob/bbb2e6b6769b03602a8ab97001f88fbec52080a1/tool_collections/fastx_toolkit/fastq_quality_boxplot/fastq_quality_boxplot.xml#L9

This would take the name from the history.

Reply:
The first part (definition of input1 and input2) is needed because if the input files of hicup_mapper does not ends with .gz it is not able to open it correctly.
The second part (trunc_result_1) is to get the file name in order to move the svg to dataset1.mapper_barchart.svg or dataset2.mapper_barchart.svg to know which plot correspond to which mate.
You are right, I think, we could use $input_first_sequence.element_identifier.

Reply:
Do you want me to do a new PR trying to simplify the code?

Reply:
If you like. I don't want to stress to you anymore, you did already such a great job here ... but I guess it would be a nice enhancement.

Reply:
In fact element_identifier give the name in the history. What I would like to get is the name of the file, for example dataset_62467.dat.
Do you know which attribute is it?

Reply:
I found: dataset_id

Reply:
Why do you want this? If it can be an arbitrary name we can also just use `./input1.fastq` and `./input2.fastq` and hardcode the names.

Reply:
Yes I realized, after...
I am doing it.